lib/hawkular/prometheus/prometheus_api.rb

```diff
@@ -0,0 +1,60 @@
+require 'hawkular/base_client'
+require 'hawkular/inventory/entities'
+require 'ostruct'
+
+module Hawkular::Prometheus
+  class Alerter < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      @entrypoint = normalize_entrypoint_url entrypoint, 'hawkular/alerter'
+      super(@entrypoint, credentials, options)
+    end
+
+    def prometheus_entrypoint
+      rest_client('/prometheus/endpoint').get
+    end
+  end
+
+  # Interface to talk with the Prometheus server used for Middleware Manager
+  # @param entrypoint [String] base url of Hawkular Services
+  class Client < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      prometheus_entrypoint = Alerter.new(entrypoint, credentials, options).prometheus_entrypoint
+      @entrypoint = normalize_entrypoint_url prometheus_entrypoint, 'api/v1'
+      super(@entrypoint, credentials, options)
+    end
+
+    def query(metrics: [], time: nil)
+      results = []
+      metrics.each do |metric|
+        query = metric['expression']
+        response = http_get "/query?start=#{time}&query=#{query}"
+        result = response['data']['result'].empty? ? {} : response['data']['result'].first
+        result['metric'] = metric
+        results << result
+      end
+      results
+    end
+
+    def query_range(metrics: [], starts: nil, ends: nil, step: nil)
```

Comment:
Looks a little like copy&paste... It's only two times; I won't be fussy. But this will deserve to be dry'ed if a third similar method comes :)

Reply:
I don't know what that means, but I guess it's a ruby thing :)  I suggest you just generate a separate PR for any ruby style enhancements you feel are necessary.

Reply:
No, it applies to any language. I mean... There is some duplicated code in `query` and `query_range`. You could have written a helper method receiving the query URL to avoid the duplicated code and avoid future double work if changes are needed :)

Reply:
ah, ok, I didn't know what "dry'ed" meant, yeah, feel free to consolidate if you're in there.